lib/src/asset/file_based.dart

```diff
@@ -46,6 +48,35 @@ class FileBasedAssetReader implements AssetReader {
       throw new InvalidInputException(id);
     }
   }
+
+  /// Searches for all [AssetId]s matching [inputSet]s.
+  Stream<AssetId> listAssetIds(Iterable<InputSet> inputSets) async* {
+    var seenAssets = new Set<AssetId>();
+    for (var inputSet in inputSets) {
+      var packageNode = packageGraph[inputSet.package];
+      var packagePath = packageNode.location.toFilePath();
+      for (var glob in inputSet.globs) {
+        var files = glob.listSync(followLinks: false, root: packagePath).where(
```

Comment:
might not matter, but I guess you could do this async, given that you are already in an async* method?